src/capture/capture_video.h

```diff
@@ -23,14 +23,155 @@
 #define DOSBOX_CAPTURE_VIDEO_H
 
 #include "render.h"
+#include "rwqueue.h"
 
-void capture_video_add_frame(const RenderedImage& image,
-                             const float frames_per_second);
+class VideoEncoder {
+public:
+	virtual void capture_video_add_frame(const RenderedImage& image,
+	                                     const float frames_per_second) = 0;
 
-void capture_video_add_audio_data(const uint32_t sample_rate,
-                                  const uint32_t num_sample_frames,
-                                  const int16_t* sample_frames);
+	virtual void capture_video_add_audio_data(const uint32_t sample_rate,
+	                                          const uint32_t num_sample_frames,
+	                                          const int16_t* sample_frames) = 0;
```

Comment:
Can we pass around std::vectors instead of raw pointers. The benefit is then you don't need to manage num_samples separately, that's less error-prone too. If too difficult, then leave it, of course. But ideally we'd only get the pointer when passing stuff to C libraries.

Reply:
This would be a nice improvement.  The ffmpeg add audio data function needs a vector anyway and this could remove a temporary copy.

I think it's a bit out of scope for this PR though.  It needs some refactoring as this same pointer goes to both the ZMBV encoder and `capture_audio.cpp`.

Reply:
Sounds reasonable, ok.